src/shell/shell.h

```diff
@@ -97,6 +98,10 @@ int flux_shell_unsetenv (flux_shell_t *shell, const char *name);
  */
 int flux_shell_get_hwloc_xml (flux_shell_t *shell, const char **xmlp);
 
+/*  Return the current shell taskmap
+ */
+const struct taskmap *flux_shell_get_taskmap (flux_shell_t *shell);
```

Comment:
This is probably a naive question but should the default taskmap be provided in string form in the plugin input args, in the same manner that the updated taskmap is set in the output args?

Not sure if there is any benefit to that.  External plugins could maybe avoid linking with libtaskmap, but how that helps I don't know.

Ignore this comment if there's nothing to be gained from it.